neutron/diffraction/convert_with_calibration.cpp

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Mads Bertelsen
+#include <boost/units/systems/si/codata/electromagnetic_constants.hpp>
+#include <boost/units/systems/si/codata/neutron_constants.hpp>
+#include <boost/units/systems/si/codata/universal_constants.hpp>
+
+#include "scipp/core/counts.h"
+#include "scipp/core/dataset.h"
+#include "scipp/core/transform.h"
+#include "scipp/core/except.h"
+#include "scipp/neutron/beamline.h"
+#include "scipp/neutron/convert.h"
+#include "scipp/neutron/diffraction/convert_with_calibration.h"
+
+using namespace scipp::core;
+
+namespace scipp::neutron {
+
+//using namespace scipp::core::except::expect;
+
+namespace diffraction {
+
+Dataset convert_with_calibration(Dataset d, const Dataset &cal){
+  // 1. check cal has the required variables
+  //scipp::core::expect::contains(cal, "tzero");
+  //scipp::core::expect::contains(cal, "difc");
```

Comment:
Why is this commented? Just remove it? Step 3.) would just fail anyway below, so we can probably avoid the explicit check?

Reply:
You mentioned that these functions would work like this in the future, so I kept it as a preparation. I believe these error messages would be nicer than if it fails in the actual algorithm.

Reply:
As mentioned in a previous review comment, those have been added by now. Furthermore, the accessors that are used have been improved and now throw a better error message. Maybe you can try and see if it is clear enough.